app/graphql/concerns/fancy_mutation.rb

```diff
@@ -0,0 +1,106 @@
+# Provides a bunch of helpers for our standard mutation structure, making consistency easy!
+#
+# @example
+#   class Mutations::Post::Like < Mutations::Base
+#     # Pass a description here to be used in the GraphQL schema
+#     description 'Like a post'
+#     # Build your input type here (set arguments)
+#     input do
+#       argument :postId, ID,
+#         required: true,
+#         description: 'The post to like'
+#     end
+#     # Your result type should generally expose what changed so clients can update their cache
+#     result Types::Post
+#     # Build the union of errors which you may return
+#     errors Types::Errors::NotAuthorized,
+#       Types::Errors::NotAuthenticated,
+#       Types::Errors::Conflict
+#
+#     def resolve(post_id:)
+#       return errors << Types::Errors::NotAuthenticated.build unless context[:current_user]
```

Comment:
would be nice if we tied this into the mutation class itself (or something via pundit) similar to how you did the `input do`.

Reply:
There's already a mixin for authenticated mutations I think, but you're right that it should get cleanly integrated into this. I also like the idea of an easy Pundit shorthand, like `authorize PostPolicy, :like?` in the class body, creating a `def ready?` hook 🤔